hazelcast-jet-core/src/main/java/com/hazelcast/jet/Sinks.java

```diff
@@ -58,6 +63,82 @@ private Sinks() {
         return new SinkImpl<>(sinkName, metaSupplier);
     }
 
+    /**
+     * Returns a sink that updates {@code Map.Entry}s it receives into a Hazelcast
+     * {@code IMap} with the specified name and update function. The function
+     * will be wrapped inside an {@link EntryProcessor}.
+     *
+     *  @param mapName name of the map
+     *  @param updateFn update function contains the update logic.
+     */
+    public static <K, V, E extends Map.Entry<K, V>> Sink<E> updateMap(
```

Comment:
We tend to use nouns for sinks. This could be called `updatingMap` ? 

There is also the case where you just want to merge the value with the new one, where they are of the same type. This is equivalent to `Collectors.toMap(Function, Function, BinaryOperator)`. This might also be a useful variation of this?